core/xchain/client_simulator/client_simulator.go

```diff
@@ -315,23 +315,31 @@ func (cs *clientSimulator) executeCheck(ctx context.Context, ruleData *deploy.IR
 	log := dlog.FromCtx(ctx).With("application", "clientSimulator")
 	log.Info("ClientSimulator executing check", "ruleData", ruleData, "cfg", cs.cfg)
 
+	requestEntitlementCheckFn := func(opts *bind.TransactOpts) (*types.Transaction, error) {
+		log.Info("Calling RequestEntitlementCheck", "opts", opts, "ruleData", ruleData)
+		gated, err := deploy.NewMockEntitlementGated(
+			cs.cfg.GetTestEntitlementContractAddress(),
+			cs.baseChain.Client,
+		)
+		if err != nil {
+			log.Error("Failed to get NewMockEntitlementGated", "err", err)
+			return nil, err
+		}
+		log.Info("NewMockEntitlementGated", "gated", gated.RequestEntitlementCheck, "err", err)
+		tx, err := gated.RequestEntitlementCheck(opts, big.NewInt(0), *ruleData)
+		log.Info("RequestEntitlementCheck called", "tx", tx, "err", err)
+		return tx, err
+	}
+
+	gasEstimate, err := cs.baseChain.TxPool.EstimateGas(ctx, requestEntitlementCheckFn)
+	log.Info("Gas estimate for request entitlement check", "gasEstimate", gasEstimate, "err", err)
+
 	pendingTx, err := cs.baseChain.TxPool.Submit(
 		ctx,
 		"RequestEntitlementCheck",
 		func(opts *bind.TransactOpts) (*types.Transaction, error) {
-			log.Info("Calling RequestEntitlementCheck", "opts", opts, "ruleData", ruleData)
-			gated, err := deploy.NewMockEntitlementGated(
-				cs.cfg.GetTestEntitlementContractAddress(),
-				cs.baseChain.Client,
-			)
-			if err != nil {
-				log.Error("Failed to get NewMockEntitlementGated", "err", err)
-				return nil, err
-			}
-			log.Info("NewMockEntitlementGated", "gated", gated.RequestEntitlementCheck, "err", err)
-			tx, err := gated.RequestEntitlementCheck(opts, big.NewInt(0), *ruleData)
-			log.Info("RequestEntitlementCheck called", "tx", tx, "err", err)
-			return tx, err
+			opts.GasLimit = max(gasEstimate, 2_500_000)
```

Comment:
It is just the simulator but is this hard coded gas limit also required when requesting a entitlement check? I thought this was only required when posting the entitlement check result to compensate for different code paths in the smart contract when a majority vote is and isn't yet reached?

Reply:
I think I can take this out actually, it was showing that the estimate actually reverted and this was due to an out of date deployment of the mock entitlement gated.